sklearn/decomposition/incremental_pca.py

```diff
@@ -188,7 +190,8 @@ def fit(self, X, y=None):
         self.singular_values_ = None
         self.noise_variance_ = None
 
-        X = check_array(X, copy=self.copy, dtype=[np.float64, np.float32])
+        X = check_array(X, accept_sparse=['csr', 'csc', 'dok', 'lil'],
```

Comment:
I don't think DOK can extract batch slices. Please test each of these cases, or just have "csr" here (or perhaps ("csr", "csc")) and the input will be converted.

Reply:
The following runs without error:
```
import numpy as np
from scipy import sparse
X = np.random.poisson(0.2, [100, 100])
for cls in [sparse.csc_matrix, sparse.csr_matrix, sparse.dok_matrix, sparse.lil_matrix]:
    cls(X)[0:10]
```

Reply:
I will add these to tests.

Reply:
It's going to be quite costly to do this slicing on DOK. Better off converting to CSR I think...